Gems/Atom/Feature/Common/Assets/Shaders/Shadow/FullscreenShadow.azsl

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+
+
+#include <Atom/Features/SrgSemantics.azsli>
+
+#include <Atom/Features/PostProcessing/FullscreenPixelInfo.azsli>
+#include <Atom/Features/PostProcessing/FullscreenVertex.azsli>
+#include <Atom/Features/PostProcessing/PostProcessUtil.azsli>
+#include <Atom/Features/ScreenSpace/ScreenSpaceUtil.azsli>
+#include <Atom/Features/Shadow/NormalOffsetShadows.azsli>
+#include <Atom/Features/Shadow/BicubicPcfFilters.azsli>
+#include <Atom/Features/Shadow/Shadow.azsli>
+#include <Atom/RPI/Math.azsli>
+#include <viewsrg.srgi>
+#include <scenesrg.srgi>
+
+
+ShaderResourceGroup PassSrg : SRG_PerPass
+{
+    Sampler PointSampler
+    {
+        MinFilter = Point;
+        MagFilter = Point;
+        MipFilter = Point;
+        AddressU = Clamp;
+        AddressV = Clamp;
+        AddressW = Clamp;
+    };
+    
+    Sampler LinearSampler
+    {
+        MinFilter = Linear;
+        MagFilter = Linear;
+        MipFilter = Linear;
+        AddressU = Clamp;
+        AddressV = Clamp;
+        AddressW = Clamp;
+    };
+
+    struct Constants
+    {
+        float2 m_screenSize;
+        int m_lightIndex;
+        int m_filterMode;
+        int m_blendBetweenCascadesEnable;        
+        int m_receiverShadowPlaneBiasEnable;
+        int m_msaaCount;
+        float m_invMsaaCount;
+    };    
+    
+    Constants m_constantData;
+    Texture2DArray<float> m_directionalLightShadowmap;   
+    Texture2DArray<float> m_directionalLightExponentialShadowmap;    
+    Texture2DMS<float> m_depth;
+}
+
+#include <Atom/RPI/ShaderResourceGroups/DefaultDrawSrg.azsli>
+#include <Atom/Features/Shadow/DirectionalLightShadowCalculator.azsli>
+
+float3 ComputeWorldNormal(VSOutput IN)
+{
+    const float offset = 1;
+    const float2 invScreenSize = rcp(PassSrg::m_constantData.m_screenSize);
+
+    const int2 texCoord = IN.m_position;
+
+    const float2 uv0 = IN.m_texCoord; // center
```

Comment:
nit: Just naming these xxxCenter, xxxRight, xxxTop would remove the need for comments and make it more clear.